### DISCUSSION TITLE
protein-protein docking interface patch constraints

### DISCUSSION THREAD

**Discussion by BrennicaMarlow** [RosettaCommons/rosetta#165] [Rosetta]
I am trying to provide constraints on the interface of two proteins. I want to constrain the docking output to dock these specific interfaces between each protein. Currently I have tried AmbiguousConstraint and KofNConstraint, but these only seem to allow you to set constraints on one protein to the entirety of another protein.

**Comment by roccomoretti** [comment]
Ambiguous and KofN constraints don't really care about which sub-constraints they use. It should totally be possible to set up an AmbiguousConstraint consisting of AtomPair constraints between one atom in one protein and one atom in another protein.

These may be tedious to set up manually, though, if you have a number of such constraints. That's why there's some convenience constraints like SiteConstraints that facilitate use cases like one-to-may interactions. But the plain Ambiguous/KofN should take manually-specified lists.

If you're using RosettaScripts, look into [Constraint Generators](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/Movers-RosettaScripts#general-movers_constraints_constraint-generators) to possibly facilitate setting up constraints.
